jena-cmds/src/main/java/tdb2/tdbcompact.java

```diff
@@ -18,18 +18,36 @@
 
 package tdb2;
 
+import org.apache.jena.base.Sys;
+import org.apache.jena.cmd.ArgDecl;
 import org.apache.jena.tdb2.store.DatasetGraphSwitchable;
 import org.apache.jena.tdb2.sys.DatabaseOps;
 import tdb2.cmdline.CmdTDB;
 
 public class tdbcompact extends CmdTDB {
+    private static final ArgDecl argDeleteOld = new ArgDecl(ArgDecl.NoValue, "deleteOld");
+
+    private boolean shouldDeleteOld = false;
+
+
     static public void main(String... argv) {
         CmdTDB.init() ;
         new tdbcompact(argv).mainRun() ;
     }
 
     protected tdbcompact(String[] argv) {
-        super(argv) ;
+        super(argv);
+
+        if (!Sys.isWindows) {
```

Comment:
It is ("should be") fine on Windows.  The compact works ... it is just the effect of delete does not happen until the JVM exits and in this command that's OK.

Reply:
fixdd